metagraph/src/annotation/taxonomy/label_to_taxid.cpp

```diff
@@ -0,0 +1,95 @@
+#include "tax_classifier.hpp"
+
+#include "common/utils/string_utils.hpp"
+
+#include "common/logger.hpp"
+
+namespace mtg {
+namespace annot {
+
+using mtg::common::logger;
+
+void TaxonomyBase::assign_label_type(const std::string &label, bool *require_accversion_to_taxid_map) {
+    if (utils::starts_with(label, "gi|")) {
+        // e.g.   >gi|1070643132|ref|NC_031224.1| Arthrobacter phage Mudcat, complete genome
+        label_type = GEN_BANK;
+        *require_accversion_to_taxid_map = true;
+    } else if (utils::starts_with(utils::split_string(label, ":")[1], "taxid|")) {
+        // e.g.   >kraken:taxid|2016032|NC_047834.1 Alteromonas virus vB_AspP-H4/4, complete genome
+        label_type = TAXID;
+        *require_accversion_to_taxid_map = false;
+    } else {
+        logger->error("Can't determine the type of the given label {}. Please make sure that the labels are in a recognized format.", label);
+        std::exit(1);
+    }
+}
```

Comment:
return bool instead of passing a pointer

Reply:
I was returning a `bool` in the previous implementation version, but thought that it could be misunderstood as a valid/failed assignation. Also, it could be a possibility to return 2 different boolean values in the future, depending on what precomputation we need to manage for different label types. 

But I can return a `bool` if you consider that it is more clear, just let me know.

Reply:
The meaning of `require_accversion_to_taxid_map` is not clear at all either.

Better return a bool and either write a short comment explaining what is returned or consider renaming the method so it's self-explanatory.

Reply:
I have changed as you suggested, thanks for the review! For the moment, I will let this comment as not solved in case of any follow-ups.